js/web/lib/wasm/jsep/webgpu/ops/attention.ts

```diff
@@ -502,35 +588,34 @@ const computeVxAttentionScore =
    if (m < uniforms.M && n < uniforms.N) {
      let outputIdx = batchIdx * uniforms.M *uniforms.v_hidden_size + m * uniforms.v_hidden_size
        + currentBatchHeadNumber * uniforms.N + n;
-     output[outputIdx] = value;
+     output[outputIdx] = output_value;
    }
   }`;
       };
 
       return context.compute(
           {
             name: 'AttentionScore',
-            shaderCache: {inputDependencies: ['type', 'type']},
-            getRunData: () => ({
-              outputs: [{dims: outputShape, dataType: probs.dataType, gpuDataType: GpuDataType.default}],
-              dispatchGroup: dispatch,
-              programUniforms
-            }),
+            shaderCache: {inputDependencies},
+            getRunData: () => ({outputs, dispatchGroup: dispatch, programUniforms}),
             getShaderSource,
           },
-          {inputs: [probs, v], outputs: [0]})[0];
+          {inputs, outputs: context.outputCount > 1 ? [0, 1] : [0]});
     };
 
 export const applyAttention =
     (context: ComputeContext, q: TensorView, k: TensorView, v: TensorView, _maskIndex: TensorView|undefined,
-     _past: TensorView|undefined, _pastKey: TensorView|undefined, _pastValue: TensorView|undefined,
+     _past: TensorView|undefined, pastKey: TensorView|undefined, pastValue: TensorView|undefined,
      relativePositionBias: TensorView|undefined, parameters: AttentionParameters, attributes: AttentionAttrs) => {
       const pastSequenceLength = context.outputCount === 1 ? 0 : parameters.pastSequenceLength;
       const totalSequenceLength = parameters.kvSequenceLength + pastSequenceLength;
-      const probs = computeAttentionProbs(
-          context, q, k, relativePositionBias, parameters, attributes, pastSequenceLength, totalSequenceLength);
+      const [probs, presentKey] = computeAttentionProbs(
+          context, q, k, pastKey, relativePositionBias, parameters, attributes, pastSequenceLength,
+          totalSequenceLength);
 
-      computeVxAttentionScore(context, probs, v, parameters, totalSequenceLength);
+      const [output, presentValue] =
+          computeVxAttentionScore(context, probs, v, pastValue, parameters, pastSequenceLength, totalSequenceLength);
+      return [output, presentKey, presentValue];
```

Comment:
I should somehow call compute to produce output instead of returning tensors here.